### DISCUSSION TITLE
pass range to :CodeCompanion

### DISCUSSION THREAD

**Discussion by Konfekt** [olimorris/codecompanion.nvim#215] [Ideas]
Seemingly :CodeCompanion only accepts the last visual selection as range, but not other ranges (`:help range`) such as the whole buffer (`:%CodeCompanion`).

**Comment by olimorris** [comment]
That's an interesting suggestion. Could you talk me through what your use case would be for supporting more than the last selection?

**Comment by Konfekt** [comment]
As I am new to this plug-in it might be that, say `%CodeCompanion` is just an alias for `:CodeCompanionChat`.
But many of the https://github.com/olimorris/codecompanion.nvim?tab=readme-ov-file#clipboard-pre-defined-prompts can apply to the whole buffer.
Now I am not sure what the default range of CodeCompanion is, but also the current line `.` might be considered, say for renaming.
Also, refactoring only the lower/upper half of the file by `.,$`/`1,.` or the pasted region `'[,']` could be a use case.

Finally, using ranges allows for operator mappings with `:help 'operatorfunc'`, so that `grip` could for example refactor the current paragraph.
